akka-remote/src/main/scala/akka/remote/artery/Handshake.scala

```diff
@@ -70,6 +70,16 @@ private[remote] class OutboundHandshake(
       private var pendingMessage: OptionVal[OutboundEnvelope] = OptionVal.None
       private var injectHandshakeTickScheduled = false
 
+      private val uniqueRemoteAddressAsyncCallback = getAsyncCallback[UniqueAddress] { _ =>
+        if (handshakeState != Completed) {
+          handshakeCompleted()
+          if (isAvailable(out))
+            pull(in)
+        }
+      }
+      private val uniqueRemoteAddressListener: UniqueAddress => Unit =
```

Comment:
perhaps add a comment that this must remain a val, because it's used in `==` when removing it `postStop`? Might be local/private enough.